### PR TITLE
Documentation: ADR for Antivirus Strategy

### PR DESCRIPTION
**Pull request by carltonsmith**
### Summary of Changes

Provides ADR for our Anti-virus strategy

Resolves https://github.com/raft-tech/TANF-app/issues/699

### CODE REVIEW

**Review by kniz-raft — APPROVED** [review]
LGTM!

**Review by jtwillis92 — APPROVED** [review]
LGTM

File: docs/Architecture Decision Record/012-antivirus-strategy.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+## Context
+
+Our original plan was to upload files to S3 using signed URLs from S3, and checking the files' headers to determine whether or not it was, in fact, a text file. Further research has revealed that there is no fool proof way of doing this and a malicious user would be able to spoof a virus as a text file.
+
+## Decision
+
+Instead of using a signed URL and sending the file directly to S3, we will instead send the file to the backend and scan it with Clam AV before sending it to S3. In the event there is a virus, we will destroy the file on the backend immediately and return an error to the frontend. 
+
+In addition to this, the frontend is able to reliably determine if a file is a binary file. The client will check submitted files for this and immediately return an error to the user. These files will not be sent to the backend.
```
Comment (alexsoble) [inline docs/Architecture Decision Record/012-antivirus-strategy.md:17]: The ClamAV decision looks great to me! As we build we may discover that some of the architectural implementation details will need to change and that's OK.

File: docs/Architecture Decision Record/012-antivirus-strategy.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+**Pros**
+
+- Less Network Requests for User
+- More timely User feedback
+  - We can now tell the user if the file is rejected before it even gets to S3
+  - Also removes the need for async notifications like email
+- More assurance that the file is not malicious
+  - ClamAV is already used for this purpose in other government projects with ATO 
+- Provides an easier foundation for future planned efforts (such as parsing file contents)
+- Compliance is more familiar with this approach
```
Comment (alexsoble) [inline docs/Architecture Decision Record/012-antivirus-strategy.md:30]: 👍 👍

**Review by alexsoble — APPROVED** [review]
LGTM!  Thanks all for the great research on this today!

Tagging you in as a reviewer @lfrohlich. The [git flow ADR](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/Architecture%20Decision%20Record/009-git-workflow.md#decision) doesn't specifically mention who needs to be tagged on ADRs, but intuitively it makes sense to have the PO approve ADRs.

**Review by lfrohlich — APPROVED** [review]
👍

File: docs/Architecture Decision Record/012-antivirus-strategy.md (hunk 1 above)
Comment (alexsoble) [inline docs/Architecture Decision Record/012-antivirus-strategy.md:17]: @carltonsmith [Not a blocker to merging!] I'm curious about the implementation for determining if a file is a binary file or not. If you have any resources or links about this, mind sending my way? Thanks!

**Pull request merged**